### DISCUSSION TITLE
Quality select for MP4 soruce

### DISCUSSION THREAD

**Discussion by ghost** [vidstack/player#864] [Q&A]
Hi, I am trying to figure out how to make mp4 quality selection in @vidstack/react

I have multiple resolutions of the same video (1080p, 720p, etc..) but I am unable to pass `<MediaPlayer>` src prop list of items of the same video that will appear in settings as quality options for` <MediaCommunitySkin />`

Is there any example how to solve this?

**Comment by mihar-22** [comment]
This feature is not available at the moment. For now we only support multiple playback qualities via HLS. We've documented it here: https://www.vidstack.io/docs/player/api/quality

**Reply by ghost** [reply]
Thank you for your answer! Is this going to be planned for the future? Also, will I be able to programmatically change the source via react state to the player without destroying or reinitializing the whole player?

This way I would be able to make another settings menu toggle items to toggle between source options (1080, 720, 480) that will trigger src state change.

**Reply by mihar-22** [reply]
Player src changes should already work but I think there might be a bug in React from a few reports. I'll look into it 👍
